tests/kafkatest/services/performance/jmx_mixin.py

```diff
@@ -0,0 +1,78 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from kafkatest.services.performance import PerformanceService
+
+
+class JmxMixin(PerformanceService):
+
+    def __init__(self, context, num_nodes, jmx_object_name=None, jmx_attributes=None):
+        super(JmxMixin, self).__init__(context, num_nodes)
+        self.jmx_object_name = jmx_object_name
+        self.jmx_attributes = jmx_attributes
+        self.jmx_port = 9192
+
+        self.started = [False] * self.num_nodes
+        self.jmx_stats = [{} for x in range(self.num_nodes)]
+        self.maximum_jmx_value = None
+        self.average_jmx_value = None
+
+    def clean_node(self, node):
+        super(JmxMixin, self).clean_node(node)
+        node.account.kill_process("jmx", clean_shutdown=False, allow_fail=True)
+        node.account.ssh("rm -rf /mnt/jmx_tool.log", allow_fail=False)
+
+    def maybe_start_jmx_tool(self, idx, node):
+        if self.started[idx-1] == True or self.jmx_object_name == None:
+            return
+        self.started[idx-1] = True
+
+        cmd = "/opt/kafka/bin/kafka-run-class.sh kafka.tools.JmxTool " \
+              "--reporting-interval 1000 --jmx-url service:jmx:rmi:///jndi/rmi://127.0.0.1:%d/jmxrmi " \
+              "--object-name %s " % (self.jmx_port, self.jmx_object_name)
+        if self.jmx_attributes != None:
+            cmd += "--attributes %s " % self.jmx_attributes
+        cmd += "> /mnt/jmx_tool.log &"
+
+        self.logger.debug("Start JmxTool %d command: %s", idx, cmd)
+        for l in node.account.ssh_capture(cmd, allow_fail=False):
```

Comment:
I just double checked it by running the test case -- simply do `node.account.ssh_capture(cmd, allow_fail=False)` without waiting for output Ducktape will not execute the cmd. I couldn't explain this though -- most likely this is related to how python execute the function with `yield`.

Can we fix it in Ducktape side so that `node.account.ssh(cmd ...)` always execute the cmd? Are you able to reproduce the problem I found?